binary_search_tree/tree.py

```diff
@@ -1,3 +1,7 @@
+from multiprocessing.dummy import current_process
+from typing import ItemsView
```

Comment:
Look like these `import` statements never got used. I'm guessing they got added automatically by autocomplete, but you can delete them regardless.